### PR TITLE
feat: support regular BuildRight+LeftAnti hash join

### LINKED ISSUE #457: Comet doesn't support Spark BroadcastHashJoinExec if it is null-aware anti-join
### What is the problem the feature request solves?

DataFusion HashJoin LeftAnti doesn't support null-aware anti join.

See https://github.com/apache/datafusion/issues/10583

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #457.

## Rationale for this change

DataFusion 53 introduced null-aware anti-join support in HashJoinExec (fixing apache/datafusion#10583), so Comet can now offload Spark's NOT IN subquery pattern (BroadcastHashJoinExec with isNullAwareAntiJoin=true) to the native layer instead of falling back to Spark. The original BuildRight + LeftAnti rejection (issue #457) was added as a workaround for that null-aware bug; with the bug fixed upstream, the rejection can be removed entirely, enabling all BuildRight + LeftAnti cases (BHJ null-aware, BHJ regular, SHJ regular) to run natively.

## What changes are included in this PR?

- Add null_aware_anti_join field to HashJoin proto and pass it to DataFusion's HashJoinExec::try_new()                                            
- Set NullAwareAntiJoin from BroadcastHashJoinExec.isNullAwareAntiJoin in the Scala serializer
- Skip swap_inputs() for null-aware anti-join (DataFusion only allows null_aware=true with LeftAnti; swap would turn it into RightAnti)           
- Remove the blanket BuildRight + LeftAnti is not supported rejection in CometHashJoin and RewriteJoin                                            
- Add BroadcastHashJoin tests for both null-aware (NOT IN subquery) and non-null-aware (BROADCAST hint with LEFT ANTI JOIN) cases; re-enable the previously-disabled SHJ BuildRight + LeftAnti test     

## How are these changes tested?

Unit tests

### CODE REVIEW

**Comment by andygrove** [comment]
Thanks @viirya! I am going to run some benchmarks at scale

**Comment by comphead** [comment]
Thanks @viirya this PR gonna improve native coverage for some TPC queries, I gonna check it this week

**Comment by mbutrovich** [comment]
Thanks @viirya for picking this up! Plan coverage jumping that much on q69 and q87 is also nice to see.

A few questions below. None feel blocking.

## Questions

Any appetite for a defensive guard in `CometHashJoin` before forwarding `bhj.isNullAwareAntiJoin`? Spark's `BroadcastHashJoinExec.scala:51-57` currently enforces `leftKeys.length == 1`, `rightKeys.length == 1`, `joinType == LeftAnti`, `buildSide == BuildRight`, and `condition.isEmpty` via `require` at construction. DataFusion enforces the first three at `HashJoinExec::try_new` (`exec.rs:405-420`). If Spark ever loosened any of those, the serde would forward `null_aware_anti_join = true` and DataFusion would reject the plan, which fails the task rather than falling back. Something like this in `operators.scala`:

```scala
val isNullAwareAntiJoin = join match {
  case bhj: BroadcastHashJoinExec => bhj.isNullAwareAntiJoin
  case _ => false
}

if (isNullAwareAntiJoin &&
    (join.leftKeys.length != 1 || join.rightKeys.length != 1 ||
     join.joinType != LeftAnti || join.buildSide != BuildRight ||
     join.condition.isDefined)) {
  withInfo(join, "null-aware anti join requires single-column LeftAnti BuildRight with no condition")
  return None
}
```

Probably overkill given how stable those invariants are, but the cost is tiny and it means Comet correctness stops depending on a Spark contract we don't own. What do you think?

A matching `plan_err!` before `HashJoinExec::try_new` in `planner.rs` would also make the failure mode clearer if we ever did hit it. DataFusion's own message is accurate but fairly generic.

## Suggested tests

Two cases I did not see covered here or in DataFusion's `test_null_aware_anti_join_*` suite.

Empty subquery. `x NOT IN (SELECT y FROM empty)` should return every left row including ones with NULL in the probe column. Distinct code path from "no NULLs and no match".

```scala
withParquetTable(Seq[(Int, Integer)]((1, 1), (2, null), (3, 3)), "tbl_a") {
  withParquetTable(Seq.empty[(Integer, Int)], "tbl_b") {
    val df = sql("SELECT * FROM tbl_a WHERE _2 NOT IN (SELECT _1 FROM tbl_b)")
    checkSparkAnswerAndOperator(df)
  }
}
```

Both sides with NULL keys. The existing tests isolate NULL to one side at a time. Since the algorithm handles build-side NULL and probe-side NULL on separate branches, might be worth one case that combines them.

```scala
withParquetTable(Seq[(Int, Integer)]((1, 1), (2, null), (3, 3)), "tbl_a") {
  withParquetTable(Seq[(Integer, Int)]((1, 100), (null, 200)), "tbl_b") {
    val df = sql("SELECT * FROM tbl_a WHERE _2 NOT IN (SELECT _1 FROM tbl_b)")
    checkSparkAnswerAndOperator(df)
  }
}
```

**Review by mbutrovich — APPROVED** [review]
Thanks @viirya! This has been a limitation for a long time! I appreciate you fixing this upstream and closing the circle here in Comet.

**Review by comphead — APPROVED** [review]
Thanks @viirya wondering should we have a separate SQL test file for this case?

**Comment by viirya** [comment]
> Thanks @viirya wondering should we have a separate SQL test file for this case?

Okay. Let me add one. Thanks.

File: spark/src/test/resources/sql-tests/join/anti_join.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+statement
+CREATE TABLE anti_right_empty(id int, k int) USING parquet
+
+-- ============================================================
+-- NOT IN subquery (null-aware anti-join)
+-- ============================================================
+
+-- Right side has no NULL: regular anti-semantics, with NULL probe filtered out
+query
```
Comment (comphead) [inline spark/src/test/resources/sql-tests/join/anti_join.sql:58]: love the test, a smallest nit: appreciate if we can move the test under `sql-tests/join`

Comment (viirya) [reply spark/src/test/resources/sql-tests/join/anti_join.sql:58]: Okay. Moved it to `sql-tests/join`.

**Comment by viirya** [comment]
Thank you @andygrove @mbutrovich @comphead
